service/src/test/kotlin/no/nav/su/se/bakover/service/søknadsbehandling/SøknadsbehandlingServiceIverksettTest.kt

```diff
@@ -314,115 +196,82 @@ internal class SøknadsbehandlingServiceIverksettTest {
                     generertDokumentJson = "",
                 ).right()
             },
-        ).let {
-            it.søknadsbehandlingService.iverksett(
-                SøknadsbehandlingService.IverksettRequest(
-                    behandlingId = avslagTilAttestering.id,
-                    attestering = attestering,
-                ),
-            ) shouldBe expectedAvslag.right()
-
-            verify(it.søknadsbehandlingRepo).hent(avslagTilAttestering.id)
-            verify(it.brevService).lagDokument(argThat { it shouldBe beOfType<Avslagsvedtak.AvslagBeregning>() })
-            verify(it.søknadsbehandlingRepo).lagre(eq(expectedAvslag), anyOrNull())
-            verify(it.vedtakRepo).lagre(argThat { it is Avslagsvedtak.AvslagBeregning }, anyOrNull())
-            verify(it.brevService).lagreDokument(
-                argThat {
-                    it.metadata.sakId shouldBe avslagTilAttestering.sakId
-                    it.metadata.vedtakId shouldNotBe null
-                    it.metadata.bestillBrev shouldBe true
-                },
-                anyOrNull()
-            )
-            verify(it.behandlingMetrics).incrementAvslåttCounter(BehandlingMetrics.AvslåttHandlinger.PERSISTERT)
-            verify(it.ferdigstillVedtakService).lukkOppgaveMedBruker(any())
-            verify(it.observer).handle(
-                argThat {
-                    it shouldBe Event.Statistikk.SøknadsbehandlingStatistikk.SøknadsbehandlingIverksatt(
-                        expectedAvslag,
-                    )
-                },
-            )
-            it.verifyNoMoreInteractions()
-        }
+        )
+
+        serviceAndMocks.søknadsbehandlingService.iverksett(
+            SøknadsbehandlingService.IverksettRequest(
+                behandlingId = avslagTilAttestering.id,
+                attestering = attestering,
+            ),
+        ) shouldBe expectedAvslag.right()
+
+        verify(serviceAndMocks.søknadsbehandlingRepo).hent(avslagTilAttestering.id)
+        verify(serviceAndMocks.brevService).lagDokument(argThat { it shouldBe beOfType<Avslagsvedtak.AvslagBeregning>() })
+        verify(serviceAndMocks.søknadsbehandlingRepo).lagre(eq(expectedAvslag), anyOrNull())
+        verify(serviceAndMocks.vedtakRepo).lagre(argThat { it is Avslagsvedtak.AvslagBeregning }, anyOrNull())
+        verify(serviceAndMocks.brevService).lagreDokument(
+            argThat {
+                it.metadata.sakId shouldBe avslagTilAttestering.sakId
+                it.metadata.vedtakId shouldNotBe null
+                it.metadata.bestillBrev shouldBe true
+            },
+            anyOrNull()
+        )
+        verify(serviceAndMocks.behandlingMetrics).incrementAvslåttCounter(BehandlingMetrics.AvslåttHandlinger.PERSISTERT)
+        verify(serviceAndMocks.ferdigstillVedtakService).lukkOppgaveMedBruker(any())
+        verify(serviceAndMocks.observer).handle(
```

Comment:
Tenker det bør være egne tester for at statistikk blir pushet, er ikke viktig for iverksetting.

Reply:
Men det er jo iverksetting per tidspunkt som sender statistikken (enig i at den bør flyttes sammen med metrikkene til asynk-jobben). Men så lenge iverksett gjør det, forventer jeg at en test feiler dersom jeg sletter den koden.

Reply:
Enig i det, men jeg tenker testen som da feiler burde vært testen "publiserer statistikk ved iverksetting" e.l. Ideelt sett hadde vi kunnet flytte/endre på statistikk-sideeffekten uten at iverksettingstesten brekker.